test/integration/quota/quota_test.go

```diff
@@ -371,3 +388,208 @@ func TestQuotaLimitedResourceDenial(t *testing.T) {
+		},
+		Spec: v1.ResourceQuotaSpec{
+			Hard: hard,
+		},
+	}
+	waitForQuota(t, quota, ctx.clientSet)
+	// create the crd instance
+	_, err := resourceClient.Create(context.TODO(), util.NewCRDInstance(definition, ns.Name, fmt.Sprintf("crd-%d", 1)), metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("failed to create crd instance 1, err: %v", err)
+	}
+
+	// confirm crd shows up in quota used count
+	err = pollExpectedQuota(ctx.clientSet, 1, resourceName, quotaName, ns.Name)
+	if err != nil {
+		t.Fatalf("err %v", err)
+	}
+
+	// delete the definition which should cascade to the instance and quota?
+	if err := apiextensionstestserver.DeleteCustomResourceDefinition(definition, ctx.apiExtensionClient); err != nil {
+		t.Fatalf("failed to delete %q: %v", definition.Name, err)
+	}
+
+	// check quota used count is now 0 after deletion
+	err = pollExpectedQuota(ctx.clientSet, 0, resourceName, quotaName, ns.Name)
+	if err != nil {
+		t.Fatalf("failed to reset quota count after CRD deletion")
+	}
+
+	// reinstall the definition
+	accessor := meta.NewAccessor()
+	accessor.SetResourceVersion(definition, "")
+	_, err = apiextensionstestserver.CreateNewCustomResourceDefinition(definition, ctx.apiExtensionClient, ctx.dynamicClient)
+	if err != nil {
+		t.Fatalf("failed to create CustomResourceDefinition on round 2: %v", err)
+	}
+
+	// create instance
+	_, err = resourceClient.Create(context.TODO(), util.NewCRDInstance(definition, ns.Name, fmt.Sprintf("crd-%d", 2)), metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("failed to create crd instance 2, err: %v", err)
+	}
+
+	// confirm quota is updated with new instance in used count
+	err = pollExpectedQuota(ctx.clientSet, 1, resourceName, quotaName, ns.Name)
+	if err != nil {
+		t.Fatalf("err %v", err)
+	}
+}
+
+func TestCRDQuota(t *testing.T) {
+	ctx := setup(t, 2)
+	defer ctx.tearDown()
+
+	// install crd on cluster, give it time to sync
+	ns := util.CreateNamespaceOrDie("test-crd", ctx.clientSet, t)
+	definition, resourceClient := util.CreateRandomCustomResourceDefinition(t, ctx.apiExtensionClient, ctx.dynamicClient, ns.Name)
+	time.Sleep(ctx.syncPeriod)
+
+	// create crd quota N
+	n := 3
+	quotaName := "quota"
+	hard := v1.ResourceList{}
+	resourceName := v1.ResourceName(fmt.Sprintf("count/%s.%s", definition.Spec.Names.Plural, definition.Spec.Group))
+	hard[resourceName] = resource.MustParse(strconv.Itoa(n))
+	quota := &v1.ResourceQuota{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      quotaName,
+			Namespace: ns.Name,
+		},
+		Spec: v1.ResourceQuotaSpec{
+			Hard: hard,
+		},
+	}
+	waitForQuota(t, quota, ctx.clientSet)
+	// create n+1 crd objects, confirm failure of the last one
+	for i := 1; i < n+1; i++ {
+		// create a new crd instance
+		_, err := resourceClient.Create(context.TODO(), util.NewCRDInstance(definition, ns.Name, fmt.Sprintf("crd-%d", i)), metav1.CreateOptions{})
+		if err != nil {
+			if i != n+1 {
+				t.Fatalf("failed to create crd on iteration: %d, err: %v", i, err)
```

Comment:
s/failed to created crd/failed to create crd instance/

Reply:
done